pkg/dependency/parser/python/poetry/parse.go

```diff
@@ -50,15 +52,16 @@ func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependenc
 	var pkgs []ftypes.Package
 	var deps []ftypes.Dependency
 	for _, pkg := range lockfile.Packages {
-		if pkg.Category == "dev" {
-			continue
-		}
-
 		pkgID := packageID(pkg.Name, pkg.Version)
 		pkgs = append(pkgs, ftypes.Package{
 			ID:      pkgID,
 			Name:    pkg.Name,
 			Version: pkg.Version,
+			// TODO upgrade logic for working with groups
+			// Mark only:
+			// - `category = "dev"`
+			// - groups without `main`. e.g. `groups = ["dev"]`
+			Dev: pkg.Category == "dev" || (len(pkg.Groups) > 0 && !slices.Contains(pkg.Groups, "main")),
```

Comment:
Checking the number of groups is not necessary, is it?
```suggestion
			Dev: pkg.Category == "dev" || !slices.Contains(pkg.Groups, "main"),
```

Reply:
I forgot to remove that after changing logic for groups. Thanks!

Reply:
It wasn't a mistake after all.
I remembered why I didn't remove that check:
it needs to check that `poetry.lock` uses groups instead of `category`.

Without that check, we'd mark all dependencies as `Dev` for `poetry.lock` with `category`:
https://github.com/aquasecurity/trivy/actions/runs/13107249603